src/pkcs11/framework-pkcs15.c

```diff
@@ -4581,6 +4692,144 @@ pkcs15_skey_get_attribute(struct sc_pkcs11_session *session,
 	return CKR_OK;
 }
 
+static CK_RV
+pkcs15_skey_unwrap(struct sc_pkcs11_session *session, void *obj,
+			CK_MECHANISM_PTR pMechanism, CK_BYTE_PTR pWrappedKey,
+			CK_ULONG ulWrappedKeyLen,
+			void *targetKey)
+{
+	struct	sc_pkcs11_card *p11card = session->slot->p11card;
+	struct	pkcs15_fw_data *fw_data = NULL;
+	struct	pkcs15_skey_object *skey = (struct pkcs15_skey_object *) obj;
+	struct	pkcs15_skey_object *targetKeyObj = (struct pkcs15_skey_object *) targetKey;
+	int	rv, flags = 0;
+
+	sc_log(context, "Initiating unwrapping with a secret key.");
+
+	fw_data = (struct pkcs15_fw_data *) p11card->fws_data[session->slot->fw_data_idx];
+	if (!fw_data)
+		return sc_to_cryptoki_error(SC_ERROR_INTERNAL, "C_UnwrapKey");
+
+	sc_log(context, "unwrapping %p %p %p %p %lu %p", session, obj,
```

Comment:
how usefull are these pointers?

Reply:
Removed the line, but added some logging in case the function fails because of a NULL argument.